components/aws/sagemaker/common/_utils.py

```diff
@@ -0,0 +1,225 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import datetime
+import os
+import subprocess
+from time import gmtime, strftime
+import time
+import string
+import random
+import json
+from urlparse import urlparse
+
+import boto3
+from botocore.exceptions import ClientError
+from sagemaker import get_execution_role
+
+def get_client(region=None):
+    """Builds a client to the AWS SageMaker API."""
+    client = boto3.client('sagemaker', region_name=region)
+    return client
+
+def get_sagemaker_role():
+  return get_execution_role()
+
+def create_training_job(client, image, data_location, output_location, role):
+  """Create a Sagemaker training job."""
+  job_name = 'TrainingJob-' + strftime("%Y%m%d%H%M%S", gmtime()) + '-' + id_generator()
+
+  create_training_params = \
+  {
+      "AlgorithmSpecification": {
+          "TrainingImage": image,
+          "TrainingInputMode": "File"
+      },
+      "RoleArn": role,
+      "OutputDataConfig": {
+          "S3OutputPath": output_location
+      },
+      "ResourceConfig": {
+          "InstanceCount": 2,
+          "InstanceType": "ml.c4.8xlarge",
+          "VolumeSizeInGB": 50
```

Comment:
Are these hard-coded params for a pipeline user?

Reply:
Yes. Right now they're hard-coded. I want to get some feedbacks and then make it more configurable in follow up PR.

Reply:
Thanks - you can assign default values

Reply:
Got you. I make it configurable now and put default value in the example.

Reply:
> Got you. I make it configurable now and put default value in the example.

BTW, you can also have default values in component.yaml:

```
{name: Some input, default: '60'}
```

Reply:
Get you. I will use this to hide some details to make examples cleaner later.